### PR TITLE
Add CircleCI config

### PR DESCRIPTION
**Pull request by dan-hill2802**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#
+# Check https://circleci.com/docs/2.0/language-python/ for more details
+#
+version: 2
+jobs:
+  build:
+    docker:
+      # specify the version you desire here
+      # use `-browsers` prefix for selenium tests, e.g. `3.6.1-browsers`
+      - image: circleci/python:2.7.15
```
Comment (mattburgess) [inline .circleci/config.yml:11]: Python-2.7 is EOL in 2020.  I'd prefer new dev to be against v3 if we can please?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          keys:
+          - v1-dependencies-{{ checksum "requirements.txt" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+
+      - run:
+          name: install dependencies
+          command: |
+            sudo pip install -r requirements.txt
+            mkdir artifacts
```
Comment (mattburgess) [inline .circleci/config.yml:34]: This looks a bit clunky.  According to the Lambda packaging docs (https://docs.aws.amazon.com/lambda/latest/dg/lambda-python-how-to-create-deployment-package.html) you should be able to do something like 'pip install -t . -r requirements.txt' to install all dependencies in the same directory as the sns_to_ses_mailer.py file and zip from the top-level of the project directory.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      docker:
+        - image: cibuilds/github:0.10
+      steps:
+        - attach_workspace:
+            at: ./artifacts
+        - run:
+            name: "Publish Release on GitHub"
+            command: |
+              ls -la artifacts
+              VERSION="v0.0.1"
```
Comment (mattburgess) [inline .circleci/config.yml:64]: I get that this is OK for now, but we probably need to find some version-bumping patterns for circleCI similar to the ones we have for concourse.

**Review by mattburgess — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (dan-hill2802) [reply .circleci/config.yml:11]: I started with some code from AWS labs and set the env to Python3, but it didn't work until I put it to 2.7. I agree that there should be a follow on piece of work to make this work in Python3 though.

File: .circleci/config.yml (hunk 2 above)
Comment (dan-hill2802) [reply .circleci/config.yml:34]: nice, will give it a go.

File: .circleci/config.yml (hunk 3 above)
Comment (dan-hill2802) [reply .circleci/config.yml:64]: agreed, well aware of this

**Comment by dan-hill2802** [comment]
How do you feel about merging this back into the org then fixing up those points as branch PRs there?

**Comment by mattburgess** [comment]
Happy with that approach, Dan.

**Pull request merged**